desc/grid.py

```diff
@@ -1243,7 +1243,9 @@ def _create_nodes(self, L=1, M=1, N=1, NFP=1):
         self._N = check_nonnegint(N, "N", False)
         self._NFP = check_posint(NFP, "NFP", False)
         self._period = (np.inf, 2 * np.pi, 2 * np.pi / self._NFP)
-        L = L + 1
+        # divide (L+1) by 2 bc only need (L+1)/2  points to
+        # integrate L-th order jacobi polynomial exactly
+        L = round((L + 1) / 2)
```

Comment:
To integrate an algebraic polynomial of degree of L exactly requires `L // 2` points (if you have complete freedom to choose quadrature points, which we do.

A lobatto grid would require `(L+2)//2`, which is what you have written if the computer performs exact arithmetic and round ups every time).

Reply:
So far the linked issue and tests show that the accuracy loss is acceptable, but are we sure that's not a characteristic unique to those tests? For example, if I multiply two polynomials together the degree doubles and you in theory need double the quadrature points to integrate it now. So when we multiply two quantities whose radial parts are given by a Jacobi polynomial of degree L in the compute functions, we would need a quadrature grid with `L` points to integrate it

Reply:
change to just `L//2`

Reply:
`(L+2)//2` is what I think we want based on this, or equivalently `L//2 + 1` I think.

we want to exactly integrate a `L` degree polynomial, want to know how many points `n` we need

Gaussian quadrature tells us `n` points lets us exactly integrate a `2n-1` order poly.

$$
\begin{align}
&L= 2n-1\\
(L+1)/2 &= n
\end{align}
$$

If `L` is odd, then this works fine and we get integer `n`.

If `L` is even, we actually get `n` being a fraction. If we floor divide, we will ignore the extra part and actually not be able to correctly integrate the `L` order poly, as we will be one short. (example, `L=2` means `(L+1)//2=1=n` which can only integrate a `2*1-1 = 1` degre polynomial.

Reply:
Yes right (L+2)//2 is interior nodes minimum. You can do L//2 if you do lobatto nodes but already know the endpoints. Got them mixed up.

Reply:
I realize this is the same as `np.ceil((L+1)/2)` but whatever I will leave as is, this will work fine